static/app/components/devtoolbar/components/issues/issuesPanel.tsx

```diff
@@ -1,21 +1,37 @@
+import {useEffect, useState} from 'react';
+
 import IssueListItem from 'sentry/components/devtoolbar/components/issueListItem';
+import {useScopeAndClient} from 'sentry/components/devtoolbar/hooks/useSentryClientAndScope';
 import Placeholder from 'sentry/components/placeholder';
 
-import useCurrentTransactionName from '../../hooks/useCurrentTransactionName';
 import {listItemPlaceholderWrapperCss} from '../../styles/listItem';
 import {panelDescCss, panelInsetContentCss, panelSectionCss} from '../../styles/panel';
 import {resetFlexColumnCss} from '../../styles/reset';
 import {smallCss} from '../../styles/typography';
 import InfiniteListItems from '../infiniteListItems';
 import InfiniteListState from '../infiniteListState';
 import PanelLayout from '../panelLayout';
+import transactionToSearchTerm from '../transactionToSearchTerm';
 
 import useInfiniteIssuesList from './useInfiniteIssuesList';
 
 export default function IssuesPanel() {
-  const transactionName = useCurrentTransactionName();
+  const {scope, client} = useScopeAndClient();
+  const [searchTerm, setSearchTerm] = useState(
+    transactionToSearchTerm(scope?.getScopeData().transactionName)
+  );
+
+  useEffect(() => {
+    if (client) {
+      client.on('startNavigationSpan', options => {
+        setSearchTerm(transactionToSearchTerm(options.name));
+      });
+    }
+  }, [client]);
```

Comment:
The docs for this specific event are [here](https://github.com/getsentry/sentry-javascript/blob/963eab78cf990e62bcf7479069c4fcb0136071a7/packages/types/src/client.ts#L288-L293)

> A hook for browser tracing integrations to trigger a span for a navigation.
> @returns A function that, when executed, removes the registered callback.

Which means that the return value of `on()` will cleanup/remove the subscription when it's called.

```suggestion
  useEffect(() => {
    if (client) {
      return client.on('startNavigationSpan', options => {
        setSearchTerm(transactionToSearchTerm(options.name));
      });
    }
    return () => {}
  }, [client]);
```